### PR TITLE
feature/CD-138: Slider component

### PR DESCRIPTION
**Pull request by gusartus**

### CODE REVIEW

File: src/common/components/Select/SelectItem.tsx [hunk 1]
```diff
@@ -15,20 +15,42 @@ interface IProps {
...
 
 	public render() {
 		const {item, isSelected} = this.props;
+		const {isHovered} = this.state;
 
 		return (
 			<div
 				className={cn(styles.selectItem, {[styles.selectItemSelected]: isSelected})}
-				onClick={this.onItemClick}
+				onMouseEnter={() => this.onItemHover()}
```
Comment (CodeAnnihilator) [inline src/common/components/Select/SelectItem.tsx]: onMouseEnter={this.onItemHover}

File: src/common/components/Select/SelectItem.tsx [hunk 2]
```diff
@@ -15,20 +15,42 @@ interface IProps {
...
+				onMouseLeave={() => this.onItemUnhover()}
```
Comment (CodeAnnihilator) [inline src/common/components/Select/SelectItem.tsx]: onMouseLeave={this.onItemUnhover}

File: src/common/components/Select/SelectItem.tsx [hunk 3]
```diff
@@ -15,20 +15,42 @@ interface IProps {
...
+		isHovered: false,
+	};
+
+	public onItemClick = (event: any) => {
+		if (!this.props.isSelected) {
+			this.props.onClick(event, this.props.item);
+		}
+	}
+
+	public onItemHover = () => this.setState({
```
Comment (CodeAnnihilator) [inline src/common/components/Select/SelectItem.tsx]: should be one function

File: src/common/components/Slider/Slider.tsx [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+	onSlide?: any;
+}
+
+export const Slider: React.FC<IProps> = ({slides, currentSlide, onSlide}) => {
+	const slidePosition = (index: number) => {
+		const diff = (index + 1) - (currentSlide + 1);
+		if (index !== currentSlide) {
+			return {
+				transform:
+					`translateX(
```
Comment (CodeAnnihilator) [inline src/common/components/Slider/Slider.tsx:20]: make a scss class and move it there. User classnames module to manage styles

File: src/common/components/Slider/Slider.tsx [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+					)`,
+			};
+		}
+	};
+
+	const navLeft = () => () => {
+		if (currentSlide > 0) onSlide(currentSlide - 1);
+	};
+
+	const navRight = () => () => {
```
Comment (CodeAnnihilator) [inline src/common/components/Slider/Slider.tsx]: this is completely wrong

File: src/common/components/Slider/Slider.tsx [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+						calc(
+							${diff * 100}%
+							${Math.sign(diff) < 0 ? ' - ' : ' + '}50px
+						)
+					)`,
+			};
+		}
+	};
+
+	const navLeft = () => () => {
```
Comment (CodeAnnihilator) [inline src/common/components/Slider/Slider.tsx]: this is completely wrong

File: src/common/components/Slider/Slider.tsx [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+								styles.slide, {
+								[styles.slide_active]: index === currentSlide,
+							})}
+							style={slidePosition(index)}
+						>
+							{slide}
+						</div>
+					))
+				}
+				<div onClick={navLeft()} className={styles.navLeftArea}>
```
Comment (CodeAnnihilator) [inline src/common/components/Slider/Slider.tsx]: this is wrong. you are calling this function each render. User callback instead

File: src/common/components/Slider/Slider.tsx [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+
+import styles from './slider.scss';
+
+interface IProps {
+	slides: any[];
+	currentSlide: number;
+	onSlide?: any;
+}
+
+export const Slider: React.FC<IProps> = ({slides, currentSlide, onSlide}) => {
```
Comment (CodeAnnihilator) [inline src/common/components/Slider/Slider.tsx]: you are already exporting this component at the end of this file

File: src/common/components/Slider/Slider.tsx (hunk 4 above)
Comment (gusartus) [reply src/common/components/Slider/Slider.tsx:20]: 15 lines of JS or 115 of SCSS? (Write correct answer below)

**Review by CodeAnnihilator — APPROVED** [review]

**Pull request merged**